src/hist1d.jl

```diff
@@ -248,6 +248,27 @@ function cumulative(h::Hist1D; forward=true)::Hist1D
     return h
 end
 
+"""
+    rebin(h::Hist1D, n::Int=1)
+    rebin(n::Int) = h::Hist1D -> rebin(h, n)
+
+Merges `n` consecutive bins into one.
+The returned histogram will have `nbins(h)/n` bins.
+"""
+function rebin(h::Hist1D, n::Int=1)
+    @assert nbins(h) % n == 0
+    p = x->Iterators.partition(x, n)
+    counts = sum.(p(bincounts(h)))
+    sumw2 = sum.(p(h.sumw2))
+    edges = first.(p(binedges(h)))
+    if _is_uniform_bins(edges)
+        s = edges[2] - first(edges)
+        edges = first(edges):s:last(edges)
```

Comment:
I wonder if this causes type instability. (i.e. re binding variable

Reply:
Yeah, looks like it's a union of vector and steprange.
```julia
julia> @code_warntype rebin(h, 2)
Variables
  #self#::Core.Const(FHist.rebin)
  h::Hist1D{Int64, Tuple{UnitRange{Int64}}}
  n::Int64
  #28::FHist.var"#28#29"{Int64}
  s::Int64
  edges::Union{StepRange{Int64, Int64}, Vector{Int64}} # <---------
  sumw2::Vector{Float64}
  counts::Vector{Int64}
  p::FHist.var"#28#29"{Int64}
```
Suggestions?